### ISSUE TITLE
Other socket types?

### ISSUE THREAD

**Issue by pwoolcoc** [rotty/zmq-tokio#2]
Is there any reason this won't work with other ZMQ socket types? I'm interested in using this with ROUTER and PUB/SUB, so let me know if there is anything I can do to help.

**Comment by vorner** [comment]
I looked through the code some time ago and AFAIK there's nothing that would prevent other types of sockets from working. I suspect the `REQ/REP` is referring to the example, not to the capabilities of the library itself.

However, I'm watching the repository and there seems to be nearly no activity from the owner, so I have no idea what the plans or state of the project are :-(. There definitely seems to be some polishing that needs doing before this is production ready.

**Comment by rotty** [comment]
Indeed, the state of the project is still in embryonic state. I've not been able to advance the state due to real-life issues, but hope to pick it up in the coming weeks.

**Comment by vorner** [comment]
That's good to hear :-).

Anyway, I have some ideas for API improvements/changes (or, at least I think these would be improvements, I guess these things are little bit subjective). Do you want to hear them? Possibly as separate issues? I'd offer to help with the actual code, but I'm afraid I'm a bit busy ATM, so it wouldn't be very soon :-(.
